### PR TITLE
chore: cancel slot migrations on shutdown

### PR DESCRIPTION
**Pull request by dranikpg**
Add ClusterFamily::Shutdown to take out all migrations

### CODE REVIEW

**Comment by dranikpg** [comment]
ok this has some issues

File: src/server/cluster/cluster_family.cc [hunk 1]
```diff
@@ -79,6 +79,27 @@ ClusterConfig* ClusterFamily::cluster_config() {
   return tl_cluster_config.get();
 }
 
+void ClusterFamily::Shutdown() {
+  vector<shared_ptr<IncomingSlotMigration>> incoming;
+  vector<shared_ptr<OutgoingMigration>> outgoing;
+  {
+    std::lock_guard lk(migration_mu_);
+    incoming = std::move(incoming_migrations_jobs_);
```
Comment (BorysTheDev) [inline src/server/cluster/cluster_family.cc:87]: I think swap will be more correct way

File: src/server/cluster/cluster_family.cc [hunk 2]
```diff
@@ -79,6 +79,21 @@ ClusterConfig* ClusterFamily::cluster_config() {
   return tl_cluster_config.get();
 }
 
+void ClusterFamily::Shutdown() {
+  shard_set->pool()->at(0)->Await([this] {
```
Comment (BorysTheDev) [inline src/server/cluster/cluster_family.cc:83]: why at 0?

**Review by BorysTheDev — APPROVED** [review]

File: src/server/cluster/cluster_family.cc (hunk 2 above)
Comment (dranikpg) [reply src/server/cluster/cluster_family.cc:83]: I didn't like this line either, but the way we initialize shards, 0..n are always shards, n..m can be connection threads

Comment (BorysTheDev) [reply src/server/cluster/cluster_family.cc:83]: Why should we run this command on the shards? I thought just run it in the same thread

**Pull request merged**